llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8911,8 +8882,9 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
       for (VPUser *U : Cur->users()) {
         auto *UserRecipe = dyn_cast<VPSingleDefRecipe>(U);
         if (!UserRecipe) {
-          assert(isa<VPLiveOut>(U) &&
-                 "U must either be a VPSingleDef or VPLiveOut");
+          assert((isa<VPLiveOut>(U) || isa<VPScalarStoreRecipe>(U)) &&
```

Comment:
better to also assert that `VPScalarStoreRecipe` is outside the vector loop region?